tests/sql/test_sql.py

```diff
@@ -309,3 +309,86 @@ def test_sql_multiple_ctes():
     expected = df1.join(df2.select(col("x").alias("a"), "y", "z"), on="a").collect().to_pydict()
 
     assert actual == expected
+
+
+def test_cast_image():
+    channels = 3
+
+    data = [
+        np.arange(4 * channels, dtype=np.uint8).reshape((2, 2, channels)),
+        np.arange(4 * channels, 13 * channels, dtype=np.uint8).reshape((3, 3, channels)),
+        None,
+    ]
+
+    s = Series.from_pylist(data, pyobj="force")
+    df = daft.from_pydict({"img": s})
+    actual = daft.sql("select cast(img as image(RGB)) from df", catalog=SQLCatalog({"df": df})).collect()
+    assert actual.schema()["img"].dtype == DataType.image("RGB")
+
+
+def test_various_type_aliases():
+    df = daft.from_pydict({"a": [1, 2, 3], "b": ["a", "b", "c"], "c": [0, 1, 0]})
+
+    actual = daft.sql("""
+    SELECT
+        cast(a as int16) as a_int16,
+        cast(a as int32) as a_int32,
+        cast(a as int64) as a_int64,
+        cast(a as uint16) as a_uint16,
+        cast(a as uint32) as a_uint32,
+        cast(a as uint64) as a_uint64,
+        cast(a as i8) as a_i8,
+        cast(a as i16) as a_i16,
+        cast(a as i32) as a_i32,
+        cast(a as i64) as a_i64,
+        cast(a as u8) as a_u8,
+        cast(a as u16) as a_u16,
+        cast(a as u32) as a_u32,
+        cast(a as u64) as a_u64,
+        cast(a as f32) as a_f32,
+        cast(a as f64) as a_f64,
+        cast(a as float32) as a_float32,
+        cast(a as float64) as a_float64,
+        cast(a as float) as a_float,
+        cast(a as real) as a_real,
+        cast(a as double) as a_double,
+        cast(b as string) as b_string,
+        cast(b as text) as b_text,
+        cast(b as varchar) as b_varchar,
+        cast(b as binary) as b_binary,
+        cast(c as bool) as c_bool,
+        cast(c as boolean) as c_boolean
```

Comment:
Perhaps this can be parameterized and mixed casing.

Reply:
keeping these as is, but added a bunch more parametrized tests in rust.